pkg/provisioner/ironic/ironic.go

```diff
@@ -1847,12 +1825,16 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 		// then the last execution of soft power off has failed.
 		if targetState == "" && ironicNode.LastError != "" {
 			p.log.Info("soft power off error", "msg", ironicNode.LastError)
-			result, _ = operationFailed(ironicNode.LastError)
-			return result, SoftPowerOffFailed{}
+			return operationFailed(ironicNode.LastError)
 		}
 		result, err = p.changePower(ironicNode, nodes.SoftPowerOff)
 		if err != nil {
-			return transientError(err)
+			switch err.(type) {
+			case HostLockedError:
+				return retryAfterDelay(powerRequeueDelay)
+			default:
+				return transientError(errors.Wrap(err, "failed to power off host"))
+			}
 		}
 		p.publisher("PowerOff", "Host soft powered off")
 	}
```

Comment:
Should we be returning operationComplete() here like hardPowerOff() does?

Reply:
Ideally, yep, good spotting. The default value of result is equivalent to what operationComplete() returns so it doesn't technically make any difference, but from a documentation perspective we should return the result of changePower inside the if statement and operationComplete() outside. I must have missed this when I originally did the result functions.